arch/risc-v/include/irq.h

```diff
@@ -803,23 +777,20 @@ noinstrument_function static inline_function bool up_interrupt_context(void)
 {
 #ifdef CONFIG_SMP
   irqstate_t flags = up_irq_save();
-#endif
-
-  bool ret = up_current_regs() != NULL;
-
-#ifdef CONFIG_SMP
+  bool ret = g_interrupt_context[up_cpu_index()];
```

Comment:
up_this_cpu()

up_cpu_index gives the physical core ID, which is not what we want here

Reply:
ok